src/combinatorics/simple_graph.lean

```diff
@@ -206,4 +206,26 @@ by { intro v, simp }
 
 end finite
 
+section maps
+
+variables {W : Type*} (G) (H : simple_graph W)
+
+/-- A graph homomorphism maps adjacent vertices to adjacent vertices -/
+abbreviation hom := rel_hom G.adj H.adj
```

Comment:
It's not clear whether this is the correct definition of `hom` yet.  The category that has lots of nice properties uses the reflexive versions of `G.adj` and `H.adj`, which allows edge contraction to induce a homomorphism.  Graph minors are subquotients in this formulation.

The reflexive version of `adj` is a relation that gives all the vertices in a neighborhood of a vertex, but the irreflexive version gives only the boundary of this neighborhood -- since graphs are sort of topological, you'd perhaps expect a definition that corresponds to continuity.

I know one nice thing about this definition is that a proper graph coloring corresponds to a homomorphism to the complete graph whose vertices are the set of colors.  But you can fix that with the other definition by requiring the homomorphism be locally injective.

Reply:
A counterpoint to what I said: a walk of length `n` seems like it ought to be a homomorphism from a path graph with n edges.  With the alternative definition, this kind of "walk" might pause at a vertex for some number of edges, so you'd need an additional local injectivity condition to match up with the traditional definition of a walk.

Reply:
To justify that category a bit, it's the category of simplicial complexes, restricted to complexes that are at most 1-dimensional.  A map of simplicial complexes is a function on vertices that sends the vertices of a simplex to the vertices of a simplex.  In particular, edges (1-simplices) might be crushed to vertices (0-simplices).

Reply:
It sounds like we once again have some graph-theoretic soul-searching to do on Zulip. I'll postpone the graph maps to another PR, because I'd like to have `rel_hom`s for my work on well-foundedness and unique factorization.